docs/content/operations/including-extensions.md

```diff
@@ -13,22 +13,137 @@ Druid extensions can be specified in the `common.runtime.properties`. There are
 
-The trick to doing this is
+2) Prepare normal extension directories under root extension directory.  Under the root extension directory, you should create sub-directories for each extension you might want to load.  Inside each sub-directory, you can put extension related files in it.
 
-1) Specify a local directory for `druid.extensions.localRepository`
+Example:
 
-2) Run the `tools pull-deps` command to pull all the specified dependencies down into your local repository
+Suppose you specify `druid.extensions.extensionsDir=/usr/local/druid/druid_extensions`, and your Druid might load normal extensions ```druid-examples```, ```druid-kafka-eight``` and ```mysql-metadata-storage```.
 
-3) Bundle up the local repository along with your other Druid stuff into whatever you use for a deployable artifact
+Then under ```druid_extensions```, it should look like this,
 
-4) Run Your druid processes with `druid.extensions.remoteRepositories=[]` and a local repository set to wherever your bundled "local" repository is located
+```
+druid_extensions/
+├── druid-examples
+│   ├── commons-beanutils-1.8.3.jar
+│   ├── commons-digester-1.8.jar
+│   ├── commons-logging-1.1.1.jar
+│   ├── commons-validator-1.4.0.jar
+│   ├── druid-examples-0.8.0-rc1.jar
+│   ├── twitter4j-async-3.0.3.jar
+│   ├── twitter4j-core-3.0.3.jar
+│   └── twitter4j-stream-3.0.3.jar
+├── druid-kafka-eight
+│   ├── druid-kafka-eight-0.7.3.jar
+│   ├── jline-0.9.94.jar
+│   ├── jopt-simple-3.2.jar
+│   ├── kafka-clients-0.8.2.1.jar
+│   ├── kafka_2.10-0.8.2.1.jar
+│   ├── log4j-1.2.16.jar
+│   ├── lz4-1.3.0.jar
+│   ├── metrics-core-2.2.0.jar
+│   ├── netty-3.7.0.Final.jar
+│   ├── scala-library-2.10.4.jar
+│   ├── slf4j-log4j12-1.6.1.jar
+│   ├── snappy-java-1.1.1.6.jar
+│   ├── zkclient-0.3.jar
+│   └── zookeeper-3.4.6.jar
+└── mysql-metadata-storage
+    ├── jdbi-2.32.jar
+    ├── mysql-connector-java-5.1.34.jar
+    └── mysql-metadata-storage-0.8.0-rc1.jar
+```
 
-The Druid processes will then only load up jars from the local repository and will not try to go out onto the internet to find the maven dependencies.
+As you can see, there are three sub-directories ```druid-examples```, ```druid-kafka-eight``` and ```mysql-metadata-storage``` under ```druid_extensions```, each sub-directory denotes an extension that Druid might load.
+
+3) Prepare hadoop dependencies directories under root hadoop dependency directory. The file structure for hadoop dependencies is almost same as normal Druid extensions described above, except that there is an extra layer of folder that specifies which version of hadoop it contains. 
+
+Example:
+
+Suppose you specify `druid.extensions.hadoopDependenciesDir=/usr/local/druid/hadoop_druid_dependencies`, and you want to prepare both `hadoop-client` 2.3.0 and 2.4.0,
+
+Then under ```hadoop_druid_dependencies```, it should look like this,
```

Comment:
we also need to add docs on how to move from the old scheme to the new scheme in the release notes.